firebase_functions/functions/src/helpers.ts

```diff
@@ -1,72 +1,63 @@
-const { createPrepareTransfers, generateAddress } = require('@iota/core');
-const { asTransactionObject } = require('@iota/transaction-converter');
+import { Client, ClientBuilder } from '@iota/client';
+import { MessageMetadata } from '@iota/client/lib/types';
 const iotaAreaCodes = require('@iota/area-codes');
 const {
   getSettings,
-  updateWalletAddressKeyIndex,
-  updateUserWalletAddressKeyIndex,
   getIotaWallet,
   getUserWallet,
   getGoogleMapsApiKey,
 } = require('./firebase');
 
-const checkRecaptcha = async (captcha, emailSettings) => {
-  const response = await axios({
-    method: 'post',
-    url: `https://www.google.com/recaptcha/api/siteverify?secret=${emailSettings.googleSecretKey}&response=${captcha}`,
-  });
-  return response ? response.data : null;
-};
+/**
+ * Initializes client with devnet node
+ * @returns Client: https://client-lib.docs.iota.org/libraries/nodejs/api_reference.html#clientbuilder
+ */
+const getClient = async (): Promise<Client> => {
+  const settings = await getSettings();
+  const client = new ClientBuilder()
+    .node(settings.provider)
+    .localPow(true)
+    .build();
+  return client
+}
 
-const getApi = async settings => {
-  const api = await composeAPI({
-    nodeWalkStrategy: new LinearWalkStrategy(
-      settings.nodes.map(provider => ({ provider }))
-    ),
-    depth: settings.tangle.depth,
-    mwm: settings.tangle.mwm,
-    successMode: SuccessMode[settings.tangle.loadBalancerSuccessMode],
-    failMode: FailMode.all,
-    timeoutMs: settings.tangle.loadBalancerTimeout,
-    // tryNodeCallback: (node) => {
-    //   console.log(`Trying node ${node.provider}`);
-    // },
-    failNodeCallback: (node, err) => {
-      console.error(`Failed node ${node.provider}, ${err.message}`);
-    }
-  });
-  
-  return api;
+/**
+ * Generates a new 32-byte (256-bit) uniformly randomly generated seed
+ * @returns Hex seed of 64 char length (32 bytes encoded in hexadecimal)
+ */
+const generateSeed = (): string => {
+  return crypto.createHash('sha256').update(crypto.randomBytes(256)).digest('hex');
 };
 
-const generateSeed = (length = 81) => {
-  const charset = 'ABCDEFGHIJKLMNOPQRSTUVWXYZ9';
-  let seed = '';
-  while (seed.length < length) {
-    const byte = crypto.randomBytes(1)
-    if (byte[0] < 243) {
-      seed += charset.charAt(byte[0] % 27);
-    }
-  }
-  return seed;
-};
+/**
+ * Generates an address with a given seed
+ * @param seed to use when generating the address
+ * @param accountIndex defaults to 0, not needed in most cases
+ * @returns address
+ */
+const generateAddress = async (seed: string, accountIndex = 0): Promise<string> => {
+  const client = await getClient()
+  const addresses = await client.getAddresses(seed)
+    .accountIndex(accountIndex)
+    .range(0, 1)
+    .get()
+  // safe assertion since includeInternal() is not used and thus addresses is a string[]
+  const address = addresses[0] as string
+  return address
+}
 
 const generateUUID = () => {
```

Comment:
Please use https://www.npmjs.com/package/uuid instead, I think we don't need to maintain our own implementation